### PR TITLE
FriendlyFirePartialFix

### PR DESCRIPTION
**Pull request by tbase9**
ICitizenColonyHandler.java
- Fixed typo

CitizenColonyHandler.java
- Fixed typo

IGuardBuilding.java
- Removed checkIfGuardShouldTakeDamage() method from since it is no longer used in EntityCitizen.java and is referenced no where else.

EntityCitizen.java
- Changed instanceof check to PlayerEntity from ServerPlayer Entity
- Changed logic in handleSourceEntityForDamage to check if PlayerEntity is from the same colony as the EntityCitizen
- Changed logic in handleSourceEntityForDamage so EntityCitizen will not take damage from Player if there is currently a raid.
- Added logic so Guards will not take damage from player if they are currently following the player and replaced with logic in EntityCitizen.java

AbstractEntityAIGuard
- Added method to check if player is the damage target when their is a raid or the guard is following the player

EntityAIKnight
- Added call to AbstractEntityAIGuard to skip applying damage if the target is a player during a raid or a player that the guard is following.

### CODE REVIEW

**Comment by tbase9** [comment]
I'm confident the Player->Guard changes should work. Less so about the Knight->Player aspect.

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java [hunk 1]
```diff
@@ -1146,4 +1147,25 @@ public boolean canBeInterrupted()
...
+    /**
+     * Method which determines whether to apply attack actions.
+     *
+     * @return the boolean of whether or not to apply attack actions.
+     */
+    public boolean handleAttackExceptions(final Entity targetEntity)
+    {
+        if (targetEntity instanceof PlayerEntity)
+        {
+            if( worker.getCitizenColonyHandler().getColony() != null && worker.getCitizenColonyHandler().getColony().getPermissions().isColonyMember((PlayerEntity) targetEntity))
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:1156]: maybe the check should here check for the guard attack permission

**Comment by Raycoms** [comment]
I think we need to catch the player damage event and then catch the player damage, and check where the damage comes from, and if from a citizen, we have to do the checks.

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java (hunk 1 above)
Comment (tbase9) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:1156]: So you want to use this method, but change the method to work differently?

**Comment by Raycoms** [comment]
Closed in favor of my own PR

**Pull request closed without merging**